src/openrct2/ride/Vehicle.cpp

```diff
@@ -6720,13 +6720,13 @@ static void AnimateMultiDimension(Vehicle& vehicle, const CarEntry& carEntry)
             else
                 vehicle.seat_rotation++;
 
-            int8_t targetSeatRotation = (vehicle.seat_rotation - 4);
+            int16_t targetSeatRotation = vehicle.seat_rotation - 4;
+            // cpp modulo allows negative values. This is the workaround.
+            while (targetSeatRotation < 0)
+                targetSeatRotation += carEntry.AnimationFrames;
+
```

Comment:
This seems like a much simpler workaround to c modulo than before.

Reply:
This is unfortunately much slower than a modulo, what exactly was the problem with negative modulo? You could always convert it to a positive with std::abs which is also almost zero cost.

Reply:
The problem is it doesn't wrap very well. The existing *should* work like this:

seat_rotation | animation_frame (8 sprites: vanilla multidim) | animation_frame (3 sprites: example)
---|---|---
0 | 4 | 2
1 | 5 | 0
2 | 6 | 1
3 | 7 | 2
4 | 0 | 0
5 | 1 |1
6 | 2|2
7 | 3|0
8 | 4|1
9 | 5|2
10 | 6|0
11 | 7|1
12 | 0|2
13 | 1|0
14 | 2|1
15 | 3|2

So I am trying to make animation_frame loop seamlessly with 4->0 being the anchor.

Doing something like:

```cpp
int16_t targetSeatRotation = std::abs((vehicle.seatRotation - 4) % carEntry.AnimationFrames);
```
will cause the values to go backwards when below 4, and doing something like:
```cpp
uint16_t targetSeatRotation = vehicle.seatRotation - 4
targetSeatRotation %= carEntry.AnimationFrames;
```
will work with powers of two, but does weird stuff when it isn't.

Reply:
Your table doesn't show any negative inputs, so this function would work for 8 sprites:
```cpp
int transform(int input) {
    int output = (input + 4) % 8;
    return output;
}```
for 3 sprites this should work:
```cpp
int transform(int input) {
    int output = (input % 3 + 2) % 3;
    return output;
}```
thats according to your truth table.

Reply:
I've pushed an update that unrolls the loop. The issue only happens when AnimationFrames is 1, 2, or 3. 1 is pointless because it's basically no animation, so 2 and 3 are the only ones that need special treatment:
```cpp
            int16_t targetSeatRotation = (vehicle.seat_rotation - 4) % carEntry.AnimationFrames;
            if (targetSeatRotation < 0)
            {
                targetSeatRotation += carEntry.AnimationFrames;
                // fix negative values when AnimationFrames is less than 4
                if (targetSeatRotation < 0)
                    targetSeatRotation += carEntry.AnimationFrames;
            }
            ```

Reply:
Not sure I fully grasp the animation stuff but if you were to add 4 instead of subtracting then you would never have any negative values, its only negative for when seat_rotation is below 4 but the wrapping is the same if you use an addition of 4 instead. Maybe the calculation should be different per animation type? I can't quite tell for sure.